Sources/SE0000_Number/Classifications/Number.swift

```diff
@@ -130,34 +130,18 @@ extension Number: AdditiveArithmetic {
     ///   - lhs: The first value to add.
     ///   - rhs: The second value to add.
     public static func + (lhs: Self, rhs: Self) -> Self {
-        switch lhs {
-            case .real(let lhsReal):
-                switch rhs {
-                    case .real(let rhsReal):
-                        return self.init(lhsReal + rhsReal)
-                    case .imaginary(let rhsImaginary):
-                        return self.init(lhsReal + rhsImaginary)
-                    case .complex(let rhsComplex):
-                        return self.init(lhsReal + rhsComplex)
-                }
-            case .imaginary(let lhsImaginary):
-                switch rhs {
-                    case .real(let rhsReal):
-                        return self.init(lhsImaginary + rhsReal)
-                    case .imaginary(let rhsImaginary):
-                        return self.init(lhsImaginary + rhsImaginary)
-                    case .complex(let rhsComplex):
-                        return self.init(lhsImaginary + rhsComplex)
-                }
-            case .complex(let lhsComplex):
-                switch rhs {
-                    case .real(let rhsReal):
-                        return self.init(lhsComplex + rhsReal)
-                    case .imaginary(let rhsImaginary):
-                        return self.init(lhsComplex + rhsImaginary)
-                    case .complex(let rhsComplex):
-                        return self.init(lhsComplex + rhsComplex)
-                }
+        switch (lhs, rhs) {
+            case let (.real(lhsReal),      .real(rhsReal     )): return self.init(lhsReal + rhsReal)
```

Comment:
This inline style, with the table-like whitespace might be a bit contraversial, but I think it makes the whole block more visually regular and easy to read.

The lines get a bit long, perhaps it's worth considering shorter variables like `lhsR`, `lhsI`, `lhsC`? If they're all the same length, then the table gets less "lumpy", too

Reply:
👍👍